src/ameba/ast/variabling/variable.cr

```diff
@@ -113,7 +113,7 @@ module Ameba::AST
     # ```
     def captured_by_block?(scope = @scope)
       scope.inner_scopes.each do |inner_scope|
-        return true if inner_scope.block? && inner_scope.references?(self)
+        return true if inner_scope.block? && inner_scope.references?(self, check_inner_scopes: false)
```

Comment:
for every **inner scope**:

- we took a variable reference
- and checked if the scope **or any inner scope** had that reference

meaning the number of iterations grows exponentially based on the number of variable references and the number of inner scopes.